rpc/jsonrpc/server/rpc_func.go

```diff
@@ -25,27 +25,38 @@ func RegisterRPCFuncs(mux *http.ServeMux, funcMap map[string]*RPCFunc, logger lo
 
 // Function introspection
 
+// FuncOptions provides the arguments for the RPCFunc
+type FuncOptions struct {
+	Ws        bool // websocket only
+	Cacheable bool // cacheable is a bool value to allow the client proxy server to cache the RPC results
+}
+
+var CacheFuncOpts = FuncOptions{
+	Cacheable: true,
+}
+
 // RPCFunc contains the introspected type information for a function
 type RPCFunc struct {
 	f        reflect.Value  // underlying rpc function
 	args     []reflect.Type // type of each function arg
 	returns  []reflect.Type // type of each return arg
 	argNames []string       // name of each argument
-	ws       bool           // websocket only
+	opts     FuncOptions    // options of the RPCFunc
 }
 
 // NewRPCFunc wraps a function for introspection.
 // f is the function, args are comma separated argument names
-func NewRPCFunc(f interface{}, args string) *RPCFunc {
-	return newRPCFunc(f, args, false)
+func NewRPCFunc(f interface{}, args string, opts FuncOptions) *RPCFunc {
```

Comment:
I think we'd generally prefer to use this kind of functional options approach when specifying options for constructors: https://github.com/tendermint/tendermint/blob/cdd3479f20b15c7dab0c683e2d5dddb7e7b95721/libs/pubsub/pubsub.go#L108

Reply:
updated with functional options, thanks for the suggestion.